src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.hpp

```diff
@@ -175,95 +328,121 @@ class TovStar : public MarkAsAnalyticSolution, public AnalyticSolution<3> {
   }
 
  protected:
-  const RadialSolution& radial_tov_solution() const;
-
-  template <typename DataType>
-  using SpatialVelocity = hydro::Tags::SpatialVelocity<DataType, 3>;
-
-  template <typename DataType>
-  using MagneticField = hydro::Tags::MagneticField<DataType, 3>;
-
-  template <typename DataType>
-  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataType>, tmpl::size_t<3>,
-                                   Frame::Inertial>;
-
-  template <typename DataType>
-  using Shift = gr::Tags::Shift<3, Frame::Inertial, DataType>;
-
-  template <typename DataType>
-  using DerivShift =
-      ::Tags::deriv<gr::Tags::Shift<3, Frame::Inertial, DataType>,
-                    tmpl::size_t<3>, Frame::Inertial>;
-
-  template <typename DataType>
-  using SpatialMetric = gr::Tags::SpatialMetric<3, Frame::Inertial, DataType>;
-
-  template <typename DataType>
-  using DerivSpatialMetric =
-      ::Tags::deriv<gr::Tags::SpatialMetric<3, Frame::Inertial, DataType>,
-                    tmpl::size_t<3>, Frame::Inertial>;
-
-  template <typename DataType>
-  using InverseSpatialMetric =
-      gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataType>;
+  const gr::Solutions::TovSolution& radial_tov_solution() const;
+
+  template <template <class, tov_detail::StarRegion> class VarsComputer,
+            typename DataType, typename... Tags, typename... VarsComputerArgs>
+  tuples::TaggedTuple<Tags...> variables_impl(
+      const tnsr::I<DataType, 3>& x, tmpl::list<Tags...> /*meta*/,
+      VarsComputerArgs&&... vars_computer_args) const {
+    const auto& radial_solution = radial_tov_solution();
+    const double outer_radius = radial_solution.outer_radius();
+    const double center_radius_cutoff = 1.e-30 * outer_radius;
+    const DataType radius = get(magnitude(x));
+    // Dispatch interior and exterior regions of the star.
```

Comment:
This adds quite a few lines of code. Given that at least in the evolution we don't really call analytic solutions very often, is it really worth the added complexity for what I expect is a relatively small performance improvement?

Reply:
The main reason I chose this implementation is to avoid repeating the iteration over grid points multiple times

Reply:
Sorry, I don't understand. Why does the `else` branch have more iterations? You just remove the `if` and `if-else` branches. That removes nearly half the lines in this function. Isn't that easier?

Reply:
Ah I understand now what you are suggesting. Sure I could just delete the `if` and `if-else` branches. The previous implementation in the ODE integration class had this logic, so I carried it over. Not sure if the performance benefit is relevant at all, on the other hand it doesn't really hurt to keep it 🤷‍♂️

Reply:
ah, I see where that code used to be. Okay, fine